--- vorbis-tools-1.2.0/configure.ac.orig	2008-03-03 06:37:31.000000000 +0100
+++ vorbis-tools-1.2.0/configure.ac	2008-05-30 19:00:51.669612156 +0200
@@ -20,11 +20,7 @@
 dnl Check for programs
 dnl --------------------------------------------------  
 
-dnl save $CFLAGS since AC_PROG_CC likes to insert "-g -O2"
-dnl if $CFLAGS is blank
-cflags_save="$CFLAGS"
 AC_PROG_CC
-CFLAGS="$cflags_save"
 
 AC_PROG_LIBTOOL
 
@@ -42,53 +38,6 @@
 dnl Set build flags based on environment
 dnl --------------------------------------------------
 
-cflags_save="$CFLAGS"
-if test -z "$GCC"; then
-        case $host in
-        *-*-irix*)
-                DEBUG="-g -signed"
-                CFLAGS="-O2 -w -signed"
-                PROFILE="-p -g3 -O2 -signed"
-                ;;
-        sparc-sun-solaris*)
-                DEBUG="-v -g"
-                CFLAGS="-xO4 -fast -w -fsimple -native -xcg92"
-                PROFILE="-v -xpg -g -xO4 -fast -native -fsimple -xcg92 -Dsuncc"
-                ;;
-        *)
-                DEBUG="-g"
-                CFLAGS="-O"
-                PROFILE="-g -p"
-                ;;
-        esac
-else
-        case $host in
-        *-*-linux*)
-                DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O2 -Wall -ffast-math -fsigned-char"
-                PROFILE="-Wall -W -pg -g -O2 -ffast-math -fsigned-char"
-                ;;
-        sparc-sun-*)
-                DEBUG="-g -Wall -fsigned-char -mv8"
-                CFLAGS="-O20 -ffast-math -fsigned-char -mv8"
-                PROFILE="-pg -g -O20 -fsigned-char -mv8"
-                ;;
-        *-*-darwin*)
-                DEBUG="-fno-common -g -Wall -fsigned-char"
-                CFLAGS="-fno-common -O4 -Wall -fsigned-char -ffast-math"
-                PROFILE="-fno-common -O4 -Wall -pg -g -fsigned-char -ffast-math"
-                ;;
-        *)
-                DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O2 -fsigned-char"
-                PROFILE="-O2 -g -pg -fsigned-char"
-                ;;
-        esac
-fi
-CFLAGS="$CFLAGS $cflags_save"
-DEBUG="$DEBUG $cflags_save"
-PROFILE="$PROFILE $cflags_save"
-
 dnl --------------------------------------------------
 dnl Allow tools to be selectively built
 dnl --------------------------------------------------
